flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/physical/batch/BatchExecExchange.scala

```diff
@@ -163,37 +163,45 @@ class BatchExecExchange(
     val inputType = input.getOutputType.asInstanceOf[BaseRowTypeInfo]
     val outputRowType = BaseRowTypeInfo.of(FlinkTypeFactory.toLogicalRowType(getRowType))
 
-    // TODO supports DataExchangeMode.BATCH in runtime
-    if (requiredExchangeMode.contains(DataExchangeMode.BATCH)) {
-      throw new TableException("DataExchangeMode.BATCH is not supported now")
+    val shuffleMode = requiredExchangeMode match {
+      case None => ShuffleMode.PIPELINED
+      case Some(mode) =>
+        mode match {
+          case DataExchangeMode.BATCH => ShuffleMode.BATCH
+          case DataExchangeMode.PIPELINED => ShuffleMode.PIPELINED
```

Comment:
how about PIPELINE_WITH_BATCH_FALLBACK?

Reply:
Now, table level never produce a `PIPELINE_WITH_BATCH_FALLBACK`.
There is a exception when `PIPELINE_WITH_BATCH_FALLBACK`.